### PR TITLE
[v2] Use kebabHash() instead of _.kebabCase()

### PR DESCRIPTION
**Pull request by chmac**
Replace uses of `_.kebabCase()` with `kebabHash()`. #4637 

This is our first Gatsby PR so all feedback welcome.

I ran `yarn test` and updated the snapshots to have the same number of test failures as the `v2` branch.

### CODE REVIEW

File: packages/gatsby-plugin-netlify/src/build-headers-program.js [hunk 1]
```diff
@@ -28,7 +29,7 @@ function linkTemplate(assetPath, type = `script`) {
 }
 
 function pathChunkName(path) {
-  const name = path === `/` ? `index` : _.kebabCase(path)
+  const name = path === `/` ? `index` : kebabHash(path)
```
Comment (pieh) [inline packages/gatsby-plugin-netlify/src/build-headers-program.js]: This actually made me realize, that netlify http/2 push most defenitely doesn't work for json data with gatsby v2, as we don't use webpack for that anymore.

**Comment by m-allanson** [comment]
Thanks @chmac! Could you merge the latest version of `v2` into this branch and get the rest of the tests passing?

**Comment by chmac** [comment]
@m-allanson When I originally submitted the PR it had the same number of failures as the v2 branch. Should the v2 branch pass all tests?

**Comment by m-allanson** [comment]
Yep, the v2 branch tests should all be passing now.

**Comment by pieh** [comment]
Finally got some time to prepare test site and indeed this fixes problem described in #4637
> In my quick and dirty testing creating a page /foo and another page /_foo results in the wrong data being supplied to /_foo.

File: packages/gatsby-plugin-netlify/src/build-headers-program.js (hunk 1 above)
Comment (pieh) [reply packages/gatsby-plugin-netlify/src/build-headers-program.js]: Re my previous comment - this isn't working correctly currently in v2, so let's not block this PR with this

**Comment by pieh** [comment]
Thanks @chmac !

**Comment by chmac** [comment]
Awesome, thanks folks, my first Gatsby contribution has landed!

**Pull request merged**